sdk/core/core-rest-pipeline/src/retryStrategies/exponentialRetryStrategy.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { PipelineResponse } from "../interfaces";
+import { RestError } from "../restError";
+import { getRandomIntegerInclusive } from "../util/helpers";
+import { RetryStrategy, RetryStrategyState } from "./retryStrategy";
+import { throttlingRetryStrategy } from "./throttlingRetryStrategy";
```

Comment:
it is strange to me that `exponentialRetryStrategy` depends on `throttlingRetryStrategy`, it feels like an unnecessary coupling of concerns. If there needs to be shared logic, can it be moved out?

Reply:
Looking ahead, this seems to be because exponential policy runs first, but we don't want it to handle throttled requests? Can't we flip the ordering of the strategies rather than make exponential policy have to be aware of this?

Reply:
I’ll look into it.